packages/noya-state/src/reducers/application.ts

```diff
@@ -1099,26 +782,92 @@ export function reducer(
 
         sharedSwatches.forEach((swatch: Sketch.Swatch) => {
           if (ids.includes(swatch.do_objectID)) {
-            swatch.name = name;
+            const newValue =
+              mode === 'replace' ? alpha : swatch.value.alpha + alpha;
+
+            swatch.value.alpha = Math.min(Math.max(0, newValue), 1);
           }
         });
       });
     }
-    case 'setSwatchOpacity': {
-      const [, id, alpha, mode = 'replace'] = action;
+    case 'addLayerStyle': {
+      return produce(state, (state) => {
+        const layerStyles = state.sketch.document.layerStyles ?? {
+          _class: 'sharedStyleContainer',
+          do_objectID: uuid(),
+          objects: [],
+        };
 
-      const ids = typeof id === 'string' ? [id] : id;
+        const style = {} as Sketch.Style;
+        const fill = {} as Sketch.Fill;
+        const contextSettings = {} as Sketch.GraphicsContextSettings;
+
+        fill.color = {
+          _class: 'color',
+          alpha: 1,
+          red: 216 / 256,
+          green: 216 / 256,
+          blue: 216 / 256,
+        };
+        fill.isEnabled = true;
+
+        contextSettings.opacity = 1;
 
+        style.fills = [fill];
+        style.contextSettings = contextSettings;
+
+        const sharedStyle: Sketch.SharedStyle = {
+          _class: 'sharedStyle',
+          do_objectID: uuid(),
+          name: 'New Style Layer',
+          value: style,
+        };
+
+        layerStyles.objects.push(sharedStyle);
+        console.log(sharedStyle);
+        state.sketch.document.layerStyles = layerStyles;
+        state.selectedLayerStyleIds = [sharedStyle.do_objectID];
+      });
+    }
+    case 'selectLayerStyle': {
+      const [, id, selectionType = 'replace'] = action;
+
+      const ids = id === undefined ? [] : typeof id === 'string' ? [id] : id;
       return produce(state, (state) => {
-        const sharedSwatches =
-          state.sketch.document.sharedSwatches?.objects ?? [];
+        switch (selectionType) {
+          case 'intersection':
+            state.selectedLayerStyleIds.push(
+              ...ids.filter((id) => !state.selectedSwatchIds.includes(id)),
+            );
+            return;
+          case 'difference':
+            ids.forEach((id) => {
+              const selectedIndex = state.selectedSwatchIds.indexOf(id);
+              state.selectedLayerStyleIds.splice(selectedIndex, 1);
+            });
+            return;
+          case 'replace':
+            state.selectedLayerStyleIds = [...ids];
+            return;
+        }
+      });
+    }
+    case 'setComponentName': {
```

Comment:
While this does result in shorter code, I actually prefer having separate actions for renaming different types. However, if it simplifies things, you could try combining them into one switch case, e.g.:

```ts
case 'setSwatchName':
case 'setSharedStyleName':
case 'setSharedTextStyleName':
case 'setSymbolName': {
  /* one implementation */
}
```

I don't mind them being separate and duplicated though, if that's more clear. We have this duplication already for border, style, shadows, etc, and I don't think it's a bad thing.